docker/mock-blockchain.Dockerfile

```
@@ -0,0 +1,39 @@
+RUN bun install --frozen-lockfile
+
+# Copy source code
+COPY apps/web ./apps/web
+COPY apps/web/lib/services/blockchain ./apps/web/lib/services/blockchain
+COPY apps/web/lib/types/blockchain ./apps/web/lib/types/blockchain
+
+# Create mock blockchain server entry point
+COPY docker/mock-blockchain-server.ts ./
+
+# Build the application
+RUN cd apps/web && bun run build
```

Hadolint (2.12.0)

[warning] Use WORKDIR to switch to a directory

(DL3003)